src/engine/scenes/visual_novel.rs

```diff
@@ -99,6 +92,35 @@ impl VisualNovel {
     }
 
 
+    fn get_next_command_by_menu_option(&self, key: Keycode) -> usize {
```

Comment:
The way that this function determines which menu choice was selected depends on the keys of the HashMap having a consistent order. However, a HashMap does not have such a guarantee. We could solve this in a few ways:
1. We could use a BTreeMap instead, which basically offers a consistent "sorting" of the entries, though this means they would need to be displayed on the screen in a sorted order, which might annoy the heck out of writers/designers.
2. We could maintain an internal mapping to the original order we displayed them in at the time the menu was rendered/created, and index against that to determine which one was selected. So the first time we display them, the order is arbitrary, but after that we can select consistently.

There might be some more options I am not thinking of.